src/java.base/share/classes/java/lang/System.java

```diff
@@ -2638,6 +2638,10 @@ public Object stringConcat1(String[] constants) {
                 return new StringConcatHelper.Concat1(constants);
             }
 
+            public byte stringInitCoder() {
```

Comment:
Why do we expose this as a coder instead of other ways, such as a `boolean hasCompactStrings()`? Are we going to use this coder elsewhere?

Reply:
Because there is already another method `stringConcatInitialCoder`, stringInitCoder is exposed to keep it consistent with the current code style as much as possible.

Reply:
I see now that you've added it in #20675. Thanks.